test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -109,8 +121,17 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
         TestableContext context;
         TearDownAgent<mi::SCX_OperatingSystem_Class_Provider>(context, CALL_LOCATION(errMsg));
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, false, context.WasRefuseUnloadCalled() );
-        system("rm -rf testChRoot > /dev/null 2>&1");
-        system("rm -rf testCWD > /dev/null 2>&1");
+        int err = system("rm -rf testChRoot > /dev/null 2>&1");
+        err = system("rm -rf testCWD > /dev/null 2>&1");
+        err = system("rm /etc/opt/microsoft/scx/conf/tmpdir > /dev/null 2>&1");
```

Comment:
You're stomping err. Err should be system() | system() | system().

That way, if anything fails, then you get a meaningful error instead of if just the last one fails.

Note, too, that you don't necessarily need this here. You can add the auto-destruct thing in specific tests to just automatically delete your test file that you created. Also, see comments above, I don't think you should be using the product path here, you should dependency inject and use that path instead, which is actually what's happening with testChRoot and testCWD, since that's using the current directory.